src/compute-client/src/controller.rs

```diff
@@ -341,8 +338,12 @@ where
             // We still have a response stashed, which we are immediately ready to process.
             return;
         }
-        if !self.replica_heartbeats.is_empty() {
-            // We have replica heartbeats waiting to be processes.
+        if !self.response_rx.is_empty() {
+            // We have responses waiting to be processed.
+            return;
+        }
```

Comment:
Why didn't we need to check this before? Was that a bug?

Reply:
Yes!